src/peerbook/libp2p_peerbook.erl

```diff
@@ -359,41 +368,33 @@ mk_this_peer(CurrentPeer, State=#state{tid=TID}) ->
         _ ->
             Associations = libp2p_peer:associations(CurrentPeer)
     end,
-    MetaDataFun = libp2p_config:get_opt(libp2p_swarm:opts(TID), [libp2p_peerbook, signed_metadata_fun],
-                                        fun() -> #{} end),
-    %% if the metadata fun crashes, simply return an empty map
-    MetaData = try MetaDataFun() of
-                   Result ->
-                       Result
-               catch
-                   _:_ -> #{}
-               end,
-    libp2p_peer:from_map(#{ pubkey => SwarmAddr,
+    {MetaData, State1} = get_signed_metadata(State),
```

Comment:
I don't quite get why you need to get metadata back from the get_signed_metadata. It's implementation _always_ returns the metadata in `#state{}`.. and that metadata is refreshed when the spawned pid returns. Why not just always pick up the metadata from state?

Reply:
Yeah, not sure why I did it this way, will rework.